### PR TITLE
feat: add new entity for ReadingStreakActions

### PR DESCRIPTION
**Pull request by ilasw**
Added new entity for `ReadingStreakActions`

### CODE REVIEW

File: src/entity/ReadingStreakActions.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  @OneToOne(() => UserStreak, (userStreak) => userStreak.userId, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  userStreak: Promise<UserStreak>;
+
+  @Column({ type: 'timestamptz', default: () => 'now()' })
+  timestamp: Date;
+
+  @Column({ type: 'enum', enum: ReadingStreakActionType })
```
Comment (rebelchris) [inline src/entity/ReadingStreakActions.ts:22]: I think we prefer storing as strings but forcing enum on graph side (this makes it a bit more flexible to add/change stuff later on)

File: src/entity/ReadingStreakActions.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Column, Entity, OneToOne, PrimaryColumn } from 'typeorm';
+import { UserStreak } from './user';
+
+export enum ReadingStreakActionType {
+  Recover = 'recovered',
+}
+
+@Entity()
+export class ReadingStreakActions {
+  @PrimaryColumn({ length: 36, default: '' })
```
Comment (rebelchris) [inline src/entity/ReadingStreakActions.ts:10]: Why is the default empty? you can just omit it on primary.
```suggestion
  @PrimaryColumn({ type: 'text' })
```

**Review by rebelchris — CHANGES_REQUESTED** [review]
You're also missing a migration for this :)

**Comment by capJavert** [comment]
@ilasw per what RFC or task are you adding this?

**Comment by rebelchris** [comment]
@capJavert This section (i also had to look)
https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/1076985865/RFC-055+-+Reading+streaks+recovery#Mutation-recoverStreak (table below)

**Comment by capJavert** [comment]
@ilasw if possible we should bundle this with some concrete queries/mutations and tests, we usually don't just add empty entities since then its easy to miss some nuance. If it helps you can make multiple smaller PRs into this and then merge in one go to `main`. Not sure how your tasks are organized.

File: src/entity/ReadingStreakActions.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  userStreak: Promise<UserStreak>;
+
+  @Column({ type: 'timestamptz', default: () => 'now()' })
+  timestamp: Date;
+
+  @Column({ type: 'text' })
+  type: string;
```
Comment (capJavert) [inline src/entity/ReadingStreakActions.ts:23]: typescript type here can still be `ReadingStreakActionType` even though in psql its `text`

Comment (capJavert) [reply src/entity/ReadingStreakActions.ts:23]: see `UserAction` for examples

**Pull request closed without merging**